src/main/java/com/candyshop/islodycze/shopping_cart/ShoppingCartController.java

```diff
@@ -38,11 +45,54 @@ public String removeShoppingCartItem(@PathVariable final Long productId,
 
         Object principal = ((UsernamePasswordAuthenticationToken) principalUser).getPrincipal();
         String username = ((UserDetails) principal).getUsername();
-        repository.deleteByUserEmailAndProductProductId(username, productId);
+        cartItemRepository.deleteByUserEmailAndProductProductId(username, productId);
 
         return "redirect:/shopping_cart";
     }
 
+    @PostMapping("/shopping_cart/add_item")
+    public String addItemsToShoppingCart(final Integer quantity, final Long productId, final Principal principalUser) {
+
+        if(quantity == null) {
+            log.error("Quantity is null");
+            return "error";
+        }
+
+        if(productId == null) {
+            log.error("productId is null");
+            return "error";
+        }
+
+        if(principalUser == null) {
+            log.info("User is null");
+            return "redirect:/login";
+        }
+
+        Long userId = getUserId((UsernamePasswordAuthenticationToken) principalUser);
+
+        CartItem duplicatedItem =
+                cartItemRepository.findByUserUserIdAndProductProductId(userId, productId);
+
+        //Prevent duplicating of items in cart
+        if (duplicatedItem == null) {
+            cartItemRepository.save(CartItem.builder()
+                                            .product(new Product().setProductId(productId))
+                                            .user(new UserEntity().setUserId(userId))
+                                            .quantity(quantity)
+                                            .build());
+        } else {
+            //TODO replace incrementation with native query
```

Comment:
I would reconsider using Update function for OrderProduct that would increment quantity, that directly changing database

Reply:
There is no update function in JpaRepository.

Reply:
I was thinking about creating a function to edit a cartItem (in CartItemService) basing on save method from JpaRepository. It 's similar to what you did already but with calculated quantity and in a service class. (Adding quantity of items in the cart with added items).